Iterable torch formatting

lhoestq opened the pull request:
Used the TorchFormatter to get torch tensors in iterable dataset with format set to "torch".

It uses the data from Arrow if possible, otherwise applies recursive_tensorize.

When set back to format_type=None, cast_to_python_objects is used.

requires https://github.com/huggingface/datasets/pull/5821

close https://github.com/huggingface/datasets/issues/5793

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/datasets/pr_5852). All of your documentation changes will be reflected on that endpoint.

lhoestq commented:
@albertvillanova could you take a look at this one ? It directly follows the arrow formatting PR

albertvillanova commented on src/datasets/iterable_dataset.py:
```diff
@@ -1118,22 +1148,25 @@ def __init__(
         shuffling: Optional[ShufflingConfig] = None,
         distributed: Optional[DistributedConfig] = None,
         token_per_repo_id: Optional[Dict[str, Union[str, bool, None]]] = None,
+        format_type="deprecated",
```
Nice you deprecate it. What about in the classes above?
- MappedExamplesIterable
- FilteredExamplesIterable

albertvillanova commented on src/datasets/iterable_dataset.py:
```diff
@@ -1118,22 +1148,25 @@ def __init__(
                 "Please pass e.g. `seed=42` in `.shuffle()` to make all the nodes use the same seed. "
             )
+        if format_type != "deprecated":
+            formatting = FormattingConfig(format_type=format_type)
```
Maybe worth raising a warning?

albertvillanova reviewed (commented):
Thanks. Some comments below.

lhoestq commented:
I added tests for the `__array__` case which lets you go from any tensor format to any other tensor format.

I also properly deprecated format_type and added a warning message.

albertvillanova reviewed (approved):
Thank you!!!
